ci-operator/step-registry/kueue-operator/install/catalog/kueue-operator-install-catalog-commands.sh

```diff
@@ -0,0 +1,209 @@
+    run_command "oc apply -f ${OC_MIRROR_OUTPUT_DIR}"
+
+    echo "[$(timestamp)] Waiting for the MachineConfigPool to finish rollout..."
+    oc wait mcp --all --for=condition=Updating --timeout=5m || true
+    oc wait mcp --all --for=condition=Updated --timeout=20m || true
+    echo "[$(timestamp)] Rollout progress completed"
+}
+
+# Applicable for 'disconnected' env
+# Note: This is a temporary workaround to avoid the disruptive impact of the 'enable-qe-catalogsource-disconnected' step.
+# As per current implementation, that step is called by every 'disconnected' cluster provisioning workflow that maintained by QE.
+# Hence this function can be removed in future once above mentioned design is well refined.
+function tmp_prune_distruptive_resource() {
+    echo "Pruning the disruptive resources in pervious step 'enable-qe-catalogsource-disconnected'..."
+    run_command "oc delete catalogsource qe-app-registry -n openshift-marketplace --ignore-not-found"
+    run_command "oc delete imagecontentsourcepolicy image-policy-aosqe --ignore-not-found"
+    run_command "oc delete imagedigestmirrorset image-policy-aosqe --ignore-not-found"
+
+    echo "[$(timestamp)] Waiting for the MachineConfigPool to finish rollout..."
+    oc wait mcp --all --for=condition=Updating --timeout=5m || true
+    oc wait mcp --all --for=condition=Updated --timeout=20m || true
+    echo "[$(timestamp)] Rollout progress completed"
+}
+
+function check_catalog_readiness () {
+    echo "Waiting the applied catalog source to become READY..."
+    if wait_for_state "catalogsource/${CATSRC_NAME}" "jsonpath={.status.connectionState.lastObservedState}=READY" "5m" "openshift-marketplace"; then
+        echo "CatalogSource is ready"
+    else
+        echo "Timed out after 5m. Dumping resources for debugging..."
+        run_command "oc get pod -n openshift-marketplace"
+        run_command "oc get event -n openshift-marketplace | grep ${CATSRC_NAME}"
+        exit 1
+    fi
+
+    echo "Storing the catalog source name to '${SHARED_DIR}/catsrc_name'..."
+    echo "${CATSRC_NAME}" > "${SHARED_DIR}"/catsrc_name
+}
+
+if [ -z "${CATSRC_IMG}" ]; then
+    echo "'CATSRC_IMG' is empty. Skipping catalog source creation..."
+    exit 0
+fi
+
+timestamp
+set_proxy
+
+# Source the environment variables from the compute step
+if test -s "${SHARED_DIR}/env" ; then
+    echo "Sourcing environment variables from compute step..."
+    source "${SHARED_DIR}/env"
+fi
+
+if [ "${MIRROR_OPERATORS}" == "true" ]; then
+    export TMP_DIR=/tmp/mirror-operators
+    export OC_MIRROR_OUTPUT_DIR="${TMP_DIR}/working-dir/cluster-resources"
+    export XDG_RUNTIME_DIR="${TMP_DIR}/run"
+    mkdir -p "${XDG_RUNTIME_DIR}/containers"
+    cd "$TMP_DIR"
+
+    check_mirror_registry
+    tmp_prune_distruptive_resource
+    configure_host_pull_secret
+    install_oc_mirror
+    mirror_catalog_and_operator
+else
+    create_catalog
```

Comment:
Just out of curiosity, in which other situation would we have MIRROR_OPERATORS as false?
Is it meant to be use by other scenarios other than disconnected?

Reply:
This is mostly for future work. Currently, we are hard-coding the values for all environments, but in the future, we might need to change them based on how we want to test things